src/shader.c

```diff
@@ -144,20 +169,9 @@ int ir_shader_new_lua(lua_State *L) {
 		return 0;
 	}
 
-	lua_createtable(L, 0, 1);
-
-	lua_pushcfunction(L, ir_shader_drop_lua);
-	lua_setfield(L, -2, "__gc");
-
-	lua_createtable(L, 0, 1);
-
-	lua_pushstring(L, "shader");
-	lua_setfield(L, -2, "__type");
-
-	lua_pushcfunction(L, ir_shader_use_lua);
-	lua_setfield(L, -2, "use");
-
-	lua_setfield(L, -2, "__index");
+    lua_getglobal(L, "ir");
+    lua_getfield(L, -1,  "shader_meta");
+    lua_replace(L, -2);
```

Comment:
I think we should standardize on ir.thing.meta to keep the top level table clean.

Reply:
Sure. But `ir.shader` is a function, unlike `ir.matrix`, which is a table, so I don't know what to do there

Reply:
We could it be like `ir.meta.thing` instead

Reply:
Shooting myself in the foot again I see... Maybe it would be better if we moved both under `ir.internal.meta.*`?

Reply:
Sounds good to me. On it